examples/api/test/material/button_style_button/button_style_button.icon_alignment.0_test.dart

```diff
@@ -0,0 +1,213 @@
+    await tester.tap(iconAlignmentStartCc);
+    await tester.pumpAndSettle();
+    expect(tester.widget<ChoiceChip>(iconAlignmentStartCc).selected, isTrue);
+    expect(tester.widget<ChoiceChip>(iconAlignmentEndCc).selected, isFalse);
+    await tester.tap(iconAlignmentEndCc);
+    await tester.pumpAndSettle();
+    expect(tester.widget<ChoiceChip>(iconAlignmentStartCc).selected, isFalse);
+    expect(tester.widget<ChoiceChip>(iconAlignmentEndCc).selected, isTrue);
+
+    final Finder textDirectionLtrCc = find.widgetWithText(ChoiceChip, 'TextDirection.ltr');
+    final Finder textDirectionRtlCc = find.widgetWithText(ChoiceChip, 'TextDirection.rtl');
+    await tester.tap(textDirectionLtrCc);
+    await tester.pumpAndSettle();
+    expect(tester.widget<ChoiceChip>(textDirectionLtrCc).selected, isTrue);
+    expect(tester.widget<ChoiceChip>(textDirectionRtlCc).selected, isFalse);
+    await tester.tap(textDirectionRtlCc);
+    await tester.pumpAndSettle();
+    expect(tester.widget<ChoiceChip>(textDirectionLtrCc).selected, isFalse);
+    expect(tester.widget<ChoiceChip>(textDirectionRtlCc).selected, isTrue);
+
+    // IconAlignment.start & TextDirection.ltr
+    await tester.tap(iconAlignmentStartCc);
+    await tester.pumpAndSettle();
+    await tester.tap(textDirectionLtrCc);
+    await tester.pumpAndSettle();
+    expect(
+      tester.widget<ElevatedButton>(
+        find.byKey(const Key('ElevatedButton.icon')),
+      ).iconAlignment,
+      IconAlignment.start,
+    );
+    expect(
+      tester.widget<FilledButton>(
+        find.byKey(const Key('FilledButton.icon')),
+      ).iconAlignment,
+      IconAlignment.start,
+    );
+    expect(
+      tester.widget<FilledButton>(
+        find.byKey(const Key('FilledButton.tonalIcon')),
+      ).iconAlignment,
+      IconAlignment.start,
+    );
+    expect(
+      tester.widget<OutlinedButton>(
+        find.byKey(const Key('OutlinedButton.icon')),
+      ).iconAlignment,
+      IconAlignment.start,
+    );
+    expect(
+      tester.widget<TextButton>(
+        find.byKey(const Key('TextButton.icon')),
+      ).iconAlignment,
+      IconAlignment.start,
+    );
+    expect(
+      tester.widget<Directionality>(
+        find.byKey(const Key('Directionality')),
+      ).textDirection,
+      TextDirection.ltr,
+    );
+
+    // IconAlignment.end & TextDirection.ltr
+    await tester.tap(iconAlignmentEndCc);
+    await tester.pumpAndSettle();
+    await tester.tap(textDirectionLtrCc);
+    await tester.pumpAndSettle();
+    expect(
+      tester.widget<ElevatedButton>(
+        find.byKey(const Key('ElevatedButton.icon')),
+      ).iconAlignment,
+      IconAlignment.end,
+    );
+    expect(
+      tester.widget<FilledButton>(
+        find.byKey(const Key('FilledButton.icon')),
+      ).iconAlignment,
+      IconAlignment.end,
+    );
+    // expect(
+    //   tester.widget<FilledButton>(
+    //     find.byKey(const Key('FilledButton.tonalIcon')),
```

Comment:
Not sure why this is failing. IconAlignment is set to IconAlignment.end & Direction is ltr. But still actuall value is IconAlignment.start. only happens for tonalIcon.